core/embed/rust/src/ui/model_mercury/component/vertical_menu.rs

```diff
@@ -0,0 +1,154 @@
+    areas_sep: AreasForSeparators,
+}
+
+impl<T> VerticalMenu<T>
+where
+    T: AsRef<str>,
+{
+    fn new(buttons: VerticalMenuButtons<T>) -> Self {
+        Self {
+            area: Rect::zero(),
+            buttons,
+            areas_sep: AreasForSeparators::new(),
+        }
+    }
+    pub fn select_word(words: [T; 3]) -> Self {
+        let mut buttons_vec = VerticalMenuButtons::new();
+        for (i, word) in words.into_iter().enumerate() {
+            let button = Button::with_text(word).styled(theme::button_vertical_menu());
+            buttons_vec.push(button);
+        }
+        Self::new(buttons_vec)
+    }
+
+    pub fn context_menu(options: [T; 3], icons: [Icon; 3]) -> Self {
+        // TODO: this is just POC
+        let mut buttons_vec = VerticalMenuButtons::new();
+        let [opt1, opt2, opt3] = options;
+        let [icon1, icon2, icon3] = icons;
+
+        buttons_vec.push(
+            Button::with_icon_and_text(IconText::new(opt1, icon1))
+                .styled(theme::button_vertical_menu()),
+        );
+        buttons_vec.push(
+            Button::with_icon_and_text(IconText::new(opt2, icon2))
+                .styled(theme::button_vertical_menu()),
+        );
+
+        buttons_vec.push(
+            Button::with_icon_and_text(IconText::new(opt3, icon3))
+                .styled(theme::button_vertical_menu_orange()),
+        );
+        Self::new(buttons_vec)
+    }
+}
+
+impl<T> Component for VerticalMenu<T>
+where
+    T: AsRef<str>,
+{
+    type Msg = VerticalMenuChoiceMsg;
+
+    fn place(&mut self, bounds: Rect) -> Rect {
+        // VerticalMenu is supposed to be used in Frame, the remaining space is just
+        // enought to fit 3 buttons separated by thin bars
+        let height_bounds_expected = 3 * MENU_BUTTON_HEIGHT + 2 * MENU_SEP_HEIGHT;
+        assert!(bounds.height() == height_bounds_expected);
+
+        self.area = bounds;
+        let mut remaining = bounds;
+        for i in 0..N_ITEMS {
+            let (area_button, new_remaining) = remaining.split_top(MENU_BUTTON_HEIGHT);
+            self.buttons[i].place(area_button);
+            remaining = new_remaining;
+            if i < N_SEPS {
+                let (area_sep, new_remaining) = remaining.split_top(MENU_SEP_HEIGHT);
+                self.areas_sep.push(area_sep);
+                remaining = new_remaining;
+            }
+        }
+
+        self.area
+    }
+
+    fn event(&mut self, ctx: &mut EventCtx, event: Event) -> Option<Self::Msg> {
+        for (i, button) in self.buttons.iter_mut().enumerate() {
+            if let Some(ButtonMsg::Clicked) = button.event(ctx, event) {
+                return Some(VerticalMenuChoiceMsg::Selected(i));
+            }
+        }
+        None
+    }
+
+    fn paint(&mut self) {
+        // TODO remove when ui-t3t1 done
+    }
+
+    fn render<'s>(&'s self, target: &mut impl Renderer<'s>) {
+        // render buttons separated by thin bars
+        for button in self.buttons.iter() {
```

Comment:
I think it's more ususal to do:
```suggestion
        for button in &self.buttons {
```